src/gtk/settings.cpp

```diff
@@ -183,40 +188,132 @@ static void notify_gtk_font_name(GObject*, GParamSpec*, void*)
 }
 }
 
-static bool UpdatePreferDark(GVariant* value)
+namespace
+{
+
+constexpr const char* TRACE_DARKMODE = "darkmode";
+
+// This function fills in the provided pointer (which must be non-null) and
+// returns GtkSettings object which is also used to indicate a failure.
+GtkSettings* GetPreferDark(gboolean* preferDarkPrev)
```

Comment:
If `preferDarkPrev` must be non-null, it seems like it should be a reference.

Reply:
We rarely use output parameters but when we do, we typically use pointers to them in order to clearly indicate that they're been modified in the caller. As I find this more readable I'd like, again, to keep it like this unless you object violently.